enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk/embed.v1.ts

```diff
@@ -0,0 +1,170 @@
+import type {
+  SdkIframeEmbedSettings,
+  SdkIframeEmbedTagMessage,
+  SdkIframeEmbedTagSettings,
+} from "./types/embed";
+
+const EMBEDDING_ROUTE = "embed/sdk/v1";
+
+type EmbedSettingKey = keyof SdkIframeEmbedSettings;
+
+const ALLOWED_EMBED_SETTING_KEYS = [
+  "apiKey",
+  "instanceUrl",
+  "dashboardId",
+  "questionId",
+  "template",
+  "theme",
+  "locale",
+] as const satisfies EmbedSettingKey[];
+
+type AllowedEmbedSettingKey = (typeof ALLOWED_EMBED_SETTING_KEYS)[number];
+
+class MetabaseEmbed {
+  static readonly VERSION = "1.0.0";
+
+  private _settings: SdkIframeEmbedTagSettings;
+  private _isEmbedReady: boolean = false;
+  private iframe: HTMLIFrameElement | null = null;
+
+  constructor(settings: SdkIframeEmbedTagSettings) {
+    if (!settings.apiKey || !settings.instanceUrl) {
+      raiseError("api key and instance url must be provided");
+    }
+
+    this._validateEmbedSettings(settings);
+    this._settings = settings;
+    this._settings._isLocalhost = this.getIsLocalhost();
+
+    this._handleMessage = this._handleMessage.bind(this);
+    this._setup();
+  }
+
+  /**
+   * Merge these settings with the current settings.
+   */
+  public updateSettings(settings: Partial<SdkIframeEmbedSettings>) {
```

Comment:
where is this used?

Reply:
@WiNloSt this is a public method that is used by the embedder in their website, which helps them to update the embed settings without reloading the whole iframe. I have an example in the end-to-end tests PR that covers this (also in the testing plan). For example, you can update the theme dynamically using this:

```tsx
embed.updateSettings({ theme: { colors: { brand: "#2d2d30" } }})
```

Reply:
Ah right. Thanks for the clarification.